src/formatter.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import {
+  FormOutlined,
+  TableOutlined,
+  FilterOutlined,
+  NumberOutlined,
+} from "@ant-design/icons";
+
+import { Schema } from "./data";
+
+export const dataFormatter = (data: Schema[]) : any[] => {
```

Comment:
I started out trying to write this with TS, but it became too complicated and eventually to move forward I just used `any`. I would not normally check in TS with `any`, because that defeats the purpose. I liked the small amount of type-checking I did have, so I left it instead of making it straight js.